tests/unit/mathematics/SGLinalg_unittest.cc

```diff
@@ -106,6 +119,23 @@ TEST(SGLinalg, GPUBackend_dot)
 	EXPECT_NEAR(result, 20.0, 1E-15);
 }
 
+TEST(SGLinalg, GPUBackend_sum)
+{
+	const index_t size=10;
+	SGVector<int32_t> a(size);
+	a.set_const(2);
+
+	GPUVector<int32_t> a_GPU(a);
+
+	std::shared_ptr<GPUBackend> ViennaCLBackend;
+	ViennaCLBackend = std::shared_ptr<GPUBackend>(new GPUBackend);
+
+	sg_linalg->set_gpu_backend(ViennaCLBackend);
```

Comment:
I suggest you keep the GPU backend test in a different file, or at least move registering the backend to a method. To avoid having to register it for every test